### PR TITLE
build on all 4 supported Travis architectures

### PR DESCRIPTION
**Pull request by mr-c**

### CODE REVIEW

**Comment by taoliu** [comment]
Thanks! I did this before to add arm64 e792658 but Travis can't build Numpy and Cython.

**Comment by mr-c** [comment]
You are welcome @taoliu !
Lets see if `sudo` fixes the problems..

**Comment by taoliu** [comment]
I am trying arm64 and see this error while compiling numpy:

https://travis-ci.com/taoliu/MACS/jobs/288971199

```
  gcc: numpy/core/src/multiarray/methods.c
  In file included from numpy/core/src/umath/loops.c.src:50:0:
  numpy/core/src/umath/loops.c.src: In function ‘LONG_right_shift’:
  numpy/core/src/umath/loops.c.src:694:47: internal compiler error: in expand_shift_1, at expmed.c:2349
       BINARY_LOOP_FAST(@type@, @type@, *out = npy_rshift@c@(in1, in2));
                                                 ^
  numpy/core/src/umath/fast_loop_macros.h:180:9: note: in definition of macro ‘BASE_BINARY_LOOP_S_INP’
           op; \
           ^
  numpy/core/src/umath/loops.c.src:694:5: note: in expansion of macro ‘BINARY_LOOP_FAST’
       BINARY_LOOP_FAST(@type@, @type@, *out = npy_rshift@c@(in1, in2));
       ^
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-5/README.Bugs> for instructions.
```

Since I saw Numpy can pass Debian autobuilder https://buildd.debian.org/status/package.php?p=numpy&suite=sid, maybe there is some specific FLAG for gcc?

**Comment by taoliu** [comment]
Now, I can test arm64 and reproduce the error

https://travis-ci.com/taoliu/MACS/builds/149693293

Add `--no-cache-dir` to `pip`: https://github.com/taoliu/MACS/blob/fix_test_multi_arch/.travis.yml

**Comment by mr-c** [comment]
Thanks @taoliu ! I got hit by a GItHub bug yesterday (my commit wasn't showing up in this PR while it was clearly visible on my branch) so I had given up for a bit.

I'll keep poking on a Python 32-bit x86 Travis CI solution.

**Comment by taoliu** [comment]
@mr-c Thanks for the update! I have tried to set 'dist: bionic' in my branch:

https://github.com/taoliu/MACS/blob/8372b149f3e41bae19f97dff7943a9fa0dc21648/.travis.yml#L1-L37 

And now MACS2 can be built under the four architectures and I can see the errors from the `cmdlinetest`.

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,53 @@
...
-  - "3.8-dev"
-matrix:
+os: linux
+dist: bionic
+jobs:
   include:
     # i386: Intel 32-bit
     - name: i386
-      language: minimal
+      language: shell
```
Comment (junaruga) [inline .travis.yml]: https://docs.travis-ci.com/user/languages/minimal-and-generic/#aliases
> Setting the language key to bash, sh or shell is equivalent to language: minimal.

Just note `language: shell` is an alias of `language: minimal` according to the document.

Comment (taoliu) [reply .travis.yml]: Yes. I changed that, together with `matrix:` to `jobs:`, `user:` to `username:`, according to the "Build config validation" information: https://travis-ci.com/taoliu/MACS/builds/150405009/config.

**Pull request merged**